### ISSUE TITLE
Address bar keeps blinking when URL is changed with replaceState

### ISSUE THREAD

**Issue by jarmo** [mozilla-mobile/firefox-ios#6375]
When URL is changed with `history.replaceState` then Firefox mobile on iOS keeps bringing back already hidden address bar, which makes UI to jump on scrolling. I suppose the reason is that viewport height will vary depending on the available visible vertical space. This problem does not happen in iOS Safari - it does keep address bar hidden until scrolled back to top.

## Steps to reproduce

Here's one simple HTML, which allows to reproduce the problem:

```html
<!DOCTYPE html>
<html>
  <head>
    <title>test</title>
    <style>
      body {
        height: 10000px;
      }
    </style>
    <script>
      var scrollTimeoutId
      window.addEventListener('scroll', function() {
        clearTimeout(scrollTimeoutId)
        scrollTimeoutId = setTimeout(changeUrl, 20)
      })

      function changeUrl() {
        history.replaceState(null, "", window.location.pathname + "?y=" + window.pageYOffset)
      }
    </script>
  </head>

  <body>
    start scrolling
  </body>
</html>
```

### Expected behavior

Address bar should be kept hidden when scrolling down.

### Actual behavior

Address bar will be made visible after scrolling down and `replaceState` changes URL making UI to "jump".

### Device & build information
* Device: iPhone 6
* Build version: N/A
### Notes
Attachments:

**Comment by dnarcese** [comment]
@jarmo Do you some examples of websites that this problem occurs on?

**Comment by jarmo** [comment]
Yes, however it is a private site (non-accessible to public), but uses the exact same approach as shown in the code above - I'm updating scroll position in the URL so that navigating away from an _infinite scroll_ page and navigating back would scroll to correct place.

Just use the html above and run it in localhost and access with mobile Firefox and this problem happens.

**Comment by jarmo** [comment]
@dnarcese added example above to my server so that anyone could see this problem easily at:
https://stash.jarmopertman.com/firefox-bug.html

Just scroll a little and wait a little to see address bar appearing.

**Comment by romaincazier** [comment]
Hi,

I have the same issue on a website I'm working on.

I'm using an IntersectionObserver to update sticky menu links to reflect the current part of the page. When replacing the history state to reflect the hash change, the address bar is made visible as described by OP.

This is working as expected on Safari.

**Comment by jarmo** [comment]
I can still reproduce this problem too on my iPhone 13 using the POC provided at https://stash.jarmopertman.com/firefox-bug.html

**Comment by YorelCayla** [comment]
I have the same issus On Iphone SE 2021 + Firefox 113.2
As soon as I stop scrooling down the page if the adresse have change the adress bar is pop's out.

This is also working as expected on Safari.

**Comment by jarmo** [comment]
Yep, problem still exists. Even on latest iOS and Firefox.

**Comment by jarmo** [comment]
Issue still there. iOS 17.6.1 and Firefox browser 130.1.

**Comment by jarmo** [comment]
Problem still exists.
